util/topgen/c_test.py

```diff
@@ -0,0 +1,94 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""This contains a class which is used to help generate top level SW tests.
+
+We maintain as a separate class rather than merge implementation with c.py,
+because both serve different purposes and need to capture the information
+differently.
+"""
+
+from typing import Dict
+
+from reggen.ip_block import IpBlock
+
+from .c import TopGenC
+from .lib import Name
+
+
+class TestPeripheral:
+    """Captures a peripheral instance's attributes for use in SW tests.
+
+    It encapsulates the settings for each peripheral's instance in the design.
+    These settings are captured as identifiers that are already defined in
+    other autogenerated / hand-edited sources such as top_{name}.h and the DIF
+    headers, rather than hard-coded constants. This is done to ensure that a
+    single source of truth is referenced in all of the tests.
+    """
+    def __init__(self, name: str, inst_name: str, base_addr_name: str):
+        self.name = name
+        self.inst_name = inst_name
+        self.base_addr_name = base_addr_name
+
+
+class IrqTestPeripheral(TestPeripheral):
```

Comment:
Is there a reason this is split into two classes? From reading this code, it's very hard to understand why it's organized in this manner. These classes just look like configuration data to me?